tests/unit/model/projections/changed-attrs-test.js

```diff
@@ -88,4 +88,191 @@ module('unit/model/projections/changed-attrs', function(hooks) {
       'changed attributes is correct'
     );
   });
+
+  test('Can set a many embedded property to a semi resolved array containing a mix of pojos and megamorphic models', function(assert) {
+    assert.expect(3);
+    this.owner.register(
+      'service:m3-schema',
+      class TestSchema extends DefaultSchema {
+        includesModel() {
+          return true;
+        }
+
+        computeNestedModel(key, value) {
+          assert.ok(true, 'Called computed nested model');
+          let attributesType;
+          try {
+            attributesType = value && value.$type;
+          } catch (e) {
+            assert.ok(false, "We don't pass Megamorphic Models to computeNestedModel");
+          }
+
+          if (key !== 'results' && typeof attributesType !== 'string') {
+            // object without a known type
+            return undefined;
+          }
```

Comment:
does this check have anything to do with the test?

Reply:
It mimics the failure mode discovered, I think we still need at least a part of this code to get it to fail robustly, but I can try and narrow down which ones.

Reply:
Fixed